docs/design/proposals/encrypted-pvc.md

```diff
@@ -0,0 +1,99 @@
+# Encrypted Persistent Volume Claims
+
+## Proposal
+Subject of this proposal is to add an encrypt feature to ceph-csi in order to allow encryption at rbd volumes level.
+
+Some but not all the benefits of this approach:
+- guarantee encryption in transit to rbd without using messenger v2
+- extra security layer to application with special regulatory needs
+- at rest encryption can be disabled to selectively allow encryption only where required
+
+## Document Terminology
+- volume encryption: encryption of a volume attached by rbd
+- encryption at rest: encryption of physical disk done by ceph
+- LUKS: Linux Unified Key Setup: stores all of the needed setup information for dm-crypt on the disk
+- dm-crypt: linux kernel device-mapper crypto target
+- cryptsetup: the command line tool to interface with dm-crypt
+
+## Proposed Solution
+
+The proposed solution in this document, is to address the volume encryption requirement by using dm-crypt module through cryptsetup cli interface.
+
+### Implementation Summary
+
+- Encryption is implemented using cryptsetup with LUKS extension.
+  A good introduction to LUKS and dm-crypt in general can be found [here](https://wiki.archlinux.org/index.php/Dm-crypt/Device_encryption#Encrypting_devices_with_cryptsetup)
+  Functions to implement necesary interaction are implemented in a separate dmcrypt.go file.
+  1. applyLUKS
```

Comment:
Nit: `createLUKS` or `initLUKS` instead of `applyLUKS`?

Reply:
👍

Reply:
initLUKS looks good